core/application/reconcile.go

```diff
@@ -118,20 +118,18 @@ func (a *app) reconcile(ctx context.Context, spec *models.MicroVM, logger *logru
 	localLogger := logger.WithField("vmid", spec.ID.String())
 	localLogger.Info("Starting reconciliation")
 
-	if spec.Status.Retry > a.cfg.MaximumRetry {
-		spec.Status.State = models.FailedState
+	plan := a.plan(spec, localLogger)
 
+	if spec.Status.Retry > a.cfg.MaximumRetry {
 		logger.Error(reachedMaximumRetryError{vmid: spec.ID, retries: spec.Status.Retry})
 
-		return nil
+		return a.saveState(ctx, spec, plan, models.FailedState)
```

Comment:
🤔  that's not what I saw in practice. I did see that initially when I had moved the `NotBefore` check above the `Retry` check, it it's current location it is fine

Reply:
But `a.saveState(ctx, spec, plan, models.FailedState)` will call `plan.Finalise(state)` and if it's a create/delete, it will be [this function](https://github.com/weaveworks/flintlock/pull/315/files#diff-65276d231339f4e6e5ca54a6487eb87916e3d4e7cdd49e4d4e3d07952ae03786R85).

Reply:
I think this check is in the right place, but the saveState is not right or the Finalise.

Reply:
🤷‍♀️  i can put those 2 settings back where they are from. it does mean we gain less from the Finalise hook, but i suppose it may be useful later if people want to add more junk

Reply:
i think i am confused by what you are saying, i just ran a check and saw no issue after the last retry attempt failed on a create, could you explain on a call?

Reply:
realised just before the call that we have a ticker which will resync and so yeh those values cannot be cleared 😅